lib/_http_client.js

```diff
@@ -115,7 +115,7 @@ function ClientRequest(options, cb) {
   var host = options.host = validateHost(options.hostname, 'hostname') ||
                             validateHost(options.host, 'host') || 'localhost';
 
-  var setHost = (options.setHost === undefined);
+  var setHost = (options.setHost === undefined || Boolean(options.setHost));
```

Comment:
Tiny nit: It would be great if you could change `var` → `let` / `const` while you are at this line.

Reply:
I think we should open another PR to fix all `var`s in this file.